### PR TITLE
Move hand animation control into grab script

### PR DESCRIPTION
**Pull request by hyperlogic**
The animations themselves are still placeholder/first pass, but this PR includes the work that creates the new hand animation states and to drive them via the handControllerGrab.js script.

Each hand now has 5 states:

* Idle - Shown briefly while the underlying body idle animation is faded in.  It should show the hand in a neutral relaxed pose.
* Open - Played when the grab script goes into 'searching' mode. 
* Grasp - Played when the grab script goes into the 'near grab' state.
* Point - Played when the grab script triggers an object, such as a light switch.
* FarGrasp - Played when the grab script goes into the 'far grab' state, and is manipulating a far away object.

The code driving the hand animations has been removed from the C++ Rig class.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 8f78a0624cd06fdf846f37246d42b4647bfe99bf available at: https://gist.github.com/91afcde688591621710f<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/pr-interface-win64-PR-6807.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6807.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 3d8f8e56182bfe42c9875c0cf0c536e09a2d4b70 available at: https://gist.github.com/91afcde688591621710f<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/pr-interface-win64-PR-6807.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6807.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: b44ae3a19798b07f057c406cb9ffebf83fcbee2f available at: https://gist.github.com/91afcde688591621710f<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/pr-interface-win64-PR-6807.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6807.dmg

File: examples/controllers/handControllerGrab.js [hunk 1]
```diff
@@ -264,6 +270,94 @@ function getSpatialOffsetRotation(hand, spatialKey) {
...
+        point: handPrefix + "HandPoint",
+        farGrasp: handPrefix + "HandFarGrasp",
+        grasp: handPrefix + "HandGrasp"
+    };
+
+    // hook up anim var handler
+    var self = this;
+    this.animHandlerId = MyAvatar.addAnimationStateHandler(function (props) {
+        return self.animStateHandler(props);
+    }, []);
```
Comment (howard-stearns) [inline examples/controllers/handControllerGrab.js]: Why not just pass self.animationStateHandler instead of wrapping it in in another function?

Comment (hyperlogic) [inline examples/controllers/handControllerGrab.js]: So when the animStateHandler function is called, this refers to an instance of HandAnimationBuddy object and not the global object.

Comment (hyperlogic) [inline examples/controllers/handControllerGrab.js]: I would have used [function.bind](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/bind) but it's not supported in this JS interpreter, so I just made a new function to close over self.

**Comment by hifi-gustavo** [comment]
Build console output for commit: fee6bda7217a88b3d89cccc9b41b3df5c00a5d99 available at: https://gist.github.com/91afcde688591621710f<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/pr-interface-win64-PR-6807.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6807.dmg

**Pull request merged**